src/Microsoft.Health.Api/Features/HealthChecks/CachedHealthCheck.cs

```diff
@@ -35,52 +35,64 @@ internal sealed class CachedHealthCheck : IHealthCheck, IDisposable
             _healthCheckFactory = EnsureArg.IsNotNull(healthCheckFactory, nameof(healthCheckFactory));
             _options = EnsureArg.IsNotNull(options, nameof(options));
             _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+            _semaphore = new SemaphoreSlim(_options.MaxRefreshThreads, _options.MaxRefreshThreads);
```

Comment:
Call looks odd to me. Passing max for both init and max.
Should we just use constructor with initialCount single param?

Reply:
It isn't necessary, but it does add a level of correctness. The only way for us to extend beyond the initial count would for an extra `Release()` invocation be made, but that could not happen without `Enter()`.